StripeFinancialConnections/StripeFinancialConnections/Source/Native/LinkAccountPicker/LinkAccountPickerViewController.swift

```diff
@@ -615,3 +676,12 @@ private func ZipAccounts(
     }
     return accountTuples
 }
+
+private func IsAccountUpdateRequired(
+    forAccount account: FinancialConnectionsPartnerAccount
+) -> Bool {
+    // repair flow
+    return account.nextPaneOnSelection == .bankAuthRepair
+    // supportability -- account requires re-sharing with additonal permissions
+    || account.nextPaneOnSelection == .partnerAuth
+}
```

Comment:
To me, this seems like a good candidate for an `extension` on `FinancialConnectionsPartnerAccount`:

```swift
private extension FinancialConnectionsPartnerAccount {
    var updateRequired: Bool {
        // repair flow
        nextPaneOnSelection == .bankAuthRepair
        // supportability -- account requires re-sharing with additional permissions
        || nextPaneOnSelection == .partnerAuth
    }
}
```

And this way it can be called as `account.updateRequired` vs `IsAccountUpdateRequired(forAccount: account)`, which to me is cleaner but it's up to you!

Reply:
yea it does look better to have an extension, my main 'worries' are around complexity:
- the logic for "is account update required" is a bit _weird_: it's very specific to this pane (and the proof for this is that I believe this _is_ the only pane where this is checked)
- similarly, one of the things I always try to do is to keep code only where its used (for example, I'll try to avoid creating a function unless its used in multiple places, I'll avoid creating an instance variable unless it needs to be referenced in multiple places etc.); this helps to "reduce complexity" because, by definition, the 'scope' of the code logic is isolated to only certain parts of the code. Of course, for this particular example, its not that much complexity, but if you multiply it by 1000's of decisions where one keeps code isolated to only one place, it starts making a difference in the long-run